pkg/volume/secret/secret_test.go

```diff
@@ -124,8 +123,7 @@ func TestPlugin(t *testing.T) {
 	doTestSecretDataInVolume(volumePath, secret, t)
 
 	metrics, err := builder.GetMetrics()
-	assert.NotEmpty(t, metrics)
-	assert.NoError(t, err)
+	volume.AssertMetricsOk(t, metrics, err)
```

Comment:
Instead of magically skipping the asserts on some builds and giving the illusion that the test passed, let's be explicit:
```
if runtime.GOOS == "darwin" || runtime.GOOS == "windows" {
	t.Skipf("not supported on GOOS=%s", runtime.GOOS)
}
```